provisioning/service/src/HttpDeviceRegistrationClient.cs

```diff
@@ -0,0 +1,660 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Provisioning.Service.Exceptions;
+
+namespace Microsoft.Azure.Devices.Provisioning.Service
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Shared;
+    using Microsoft.Azure.Devices.Common;
+    using Microsoft.Azure.Devices.Common.Exceptions;
+
+    class HttpDeviceRegistrationClient : DeviceRegistrationClient, IDisposable
+    {
+        const string EnrollmentUriFormat = "enrollments/{0}?{1}";
+        const string BulkEnrollmentUriFormat = "enrollments?{0}";
+        const string EnrollmentGroupUriFormat = "enrollmentGroups/{0}?{1}";
+        const string QueryEnrollmentUriFormat = "enrollments/query?{0}";
+        const string QueryEnrollmentGroupUriFormat = "enrollmentGroups/query?{0}";
+        const string DeviceRegistrationUriFormat = "registrations/{0}?{1}";
+        const string DeviceRegistrationsUriFormat = "registrations/{0}/query?{1}";
+
+        const string ContinuationTokenHeader = "x-ms-continuation";
+        const string PageSizeHeader = "x-ms-max-item-count";
+
+        private const string ApiVersionQueryString = ClientApiVersionHelper.ApiVersionQueryString;
+
+        static readonly Regex DeviceIdRegex = new Regex(@"^[A-Za-z0-9\-:.+%_#*?!(),=@;$']{1,128}$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+        static readonly TimeSpan DefaultOperationTimeout = TimeSpan.FromSeconds(100);
+
+        IHttpClientHelper httpClientHelper;
+        readonly string drsName;
+
+        internal HttpDeviceRegistrationClient(IotHubConnectionString connectionString)
+        {
+            this.drsName = connectionString.IotHubName;
+            this.httpClientHelper = new HttpClientHelper(
+                connectionString.HttpsEndpoint,
+                connectionString,
+                ExceptionHandlingHelper.GetDefaultErrorMapping(),
+                DefaultOperationTimeout,
+                client => { });
+        }
+
+        // internal test helper
+        internal HttpDeviceRegistrationClient(IHttpClientHelper httpClientHelper, string drsName)
```

Comment:
removed and we will need to figure out another way to test this instead of migrating the exisiting tests over.

---
In reply to: [141720270](https://github.com/Azure/azure-iot-sdk-csharp/pull/174#discussion_r141720270) [](ancestors = 141720270)